src/hierarchical_interleaved_bloom_filter.cpp

```diff
@@ -46,16 +46,24 @@ size_t hierarchical_build(hierarchical_interleaved_bloom_filter & hibf,
 {
     size_t const ibf_pos{data.request_ibf_idx()};
 
-    std::vector<uint64_t> ibf_positions(current_node.number_of_technical_bins, ibf_pos);
-    std::vector<uint64_t> filename_indices(current_node.number_of_technical_bins, bin_kind::merged);
+    auto & next_ibf_id = hibf.next_ibf_id[ibf_pos];
```

Comment:
```suggestion
    auto & technical_bin_to_ibf_id = hibf.next_ibf_ids[ibf_pos];
```
? 
It's again a vector over every technical bin, no?